File: FleetManager.Domain/Entities/Vehicle.cs
namespace FleetManager.Domain.Entities
{
    public enum VehicleStatus
    {
        Active,
        Inactive,
        Maintenance,
        Retired
    }

    public class Vehicle
    {
        public Guid Id { get; set; }
        public string PlateNumber { get; set; } = null!;
        public string Make { get; set; } = null!;
        public string Model { get; set; } = null!;
        public int Year { get; set; }

        // Estado actual del vehículo
        public VehicleStatus Status { get; set; } = VehicleStatus.Active;

        // Kilometraje actual
        public int Kilometers { get; set; } = 0;

        // Número de bastidor (VIN)
        public string VIN { get; set; } = null!;

        // Fecha de adquisición o registro en sistema
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Fecha de última revisión técnica (ITV)
        public DateTime? LastInspectionDate { get; set; }

        // Fecha próxima revisión técnica
        public DateTime? NextInspectionDue { get; set; }

        // Fecha de la última actualización de estado o kilometraje
        public DateTime? LastUpdated { get; set; }

        // Opcional: ubicación actual
        public string? Location { get; set; }

        // Opcional: notas o comentarios libres
        public string? Notes { get; set; }
    }
}

File: fleetmanager-frontend/src/types/Vehicle.ts
export const vehicleStatuses = ["Active", "Inactive", "Maintenance", "Retired"] as const;
export type VehicleStatus = typeof vehicleStatuses[number];

export interface Vehicle {
	id: string;
	plateNumber: string;
	make: string;
	model: string;
	year: number;
	status: VehicleStatus;
	kilometers: number;
	vin: string;
	createdAt: string;
	lastInspectionDate?: string;
	nextInspectionDue?: string;
	lastUpdated?: string;
	location?: string;
	notes?: string;
}

export interface VehiclesSummary {
	total: number;
	active: number;
	inactive: number;
	maintenance: number;
	retired: number;
}

export interface VehicleListFilters {
	page: number;
	pageSize?: number;
	status?: VehicleStatus;
	search?: string;
	sortBy?: string;
	sortDesc?: boolean;
}